v2/writer.go

```diff
@@ -130,3 +131,65 @@ func (w *Writer) writeIndex(writer io.Writer, carV1 []byte) (int64, error) {
 	// FIXME refactor index to expose the number of bytes written.
 	return 0, index.WriteTo(idx, writer)
 }
+
+// WrapV1File takes a source path to a CARv1 file and wraps it as a CARv2 file
+// with an index, writing the result to the destination path.
+// The resulting CARv2 file's inner CARv1 payload is left unmodified,
+// and does not use any padding before the innner CARv1 or index.
+func WrapV1File(srcPath, dstPath string) error {
+	// TODO: verify src is indeed a CARv1 to prevent misuse.
+	// index.Generate should probably be in charge of that.
+
+	// TODO: also expose WrapV1(io.ReadSeeker, io.Writer),
+	// once index.Generate takes a ReadSeeker.
+
+	// We don't use mmap.Open, so we can later use io.Copy.
+	f1, err := os.Open(srcPath)
+	if err != nil {
+		return err
+	}
+	defer f1.Close()
```

Comment:
Should we check the err return value of f1.Close()?

Reply:
I don't think so, given that we only open it for reading.